src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -378,20 +386,21 @@ await foreach (var response in requests)
             var csharpTextDocument = semanticTokensParams.TextDocument with { Uri = csharpDoc.Uri };
             semanticTokensParams = semanticTokensParams with { TextDocument = csharpTextDocument };
 
-            var newParams = new SemanticTokensParams
+            var newParams = new SemanticTokensRangeParams
```

Comment:
How does the `semanticTokensParams.Range` make it onto this new params object?

Reply:
It makes it onto the object via [this line](https://github.com/dotnet/razor-tooling/blob/aca8c36264aec6e023c273a07faf30f323765086/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs#L417) in the semantic tokens service. In my latest push I'm also changing `Range` to be a required parameter after feedback from Ryan

Reply:
Hmmm, maybe I'm reading into this wrong but don't we drop the range accidentally prior to re-invoking? Aka we construct this "newParams" and then pass it to reinvoke below but I'm not seeing where we update the `newParams` to lift the `range` from `semanticTokensParams`